staging/src/k8s.io/apimachinery/BUILD

```diff
@@ -0,0 +1,64 @@
+filegroup(
+    name = "package-srcs",
+    srcs = glob(["**"]),
+    tags = ["automanaged"],
+    visibility = ["//visibility:private"],
+)
+
+filegroup(
+    name = "all-srcs",
+    srcs = [
+        ":package-srcs",
+        "//staging/src/k8s.io/apimachinery/pkg/api/apitesting:all-srcs",
```

Comment:
Does this mean that sub-directory hierarchy changes will required top-level apimachinery approval from now on? :-/

Reply:
Well, looking at the diff above, it was even worse before.

Reply:
That's correct. The easiest way to fix it is to ensure there's a `BUILD` file at appropriate ownership levels, since the `all-srcs` rules fold up to the next `all-srcs` rule.

For example, the `pkg/apis/OWNERS` file here has
```yaml
options:
  no_parent_owners: true
approvers:
- api-approvers
reviewers:
- api-reviewers
labels:
- kind/api-change
```

and since there are no `BUILD` files until well into the tree, the directory structure is encoded here in this top-level `BUILD` file:
```python
        "//staging/src/k8s.io/apimachinery/pkg/apis/meta/fuzzer:all-srcs",
        "//staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion:all-srcs",
        "//staging/src/k8s.io/apimachinery/pkg/apis/meta/v1:all-srcs",
        "//staging/src/k8s.io/apimachinery/pkg/apis/meta/v1beta1:all-srcs",
        "//staging/src/k8s.io/apimachinery/pkg/apis/testapigroup:all-srcs",
```

If we were to create a `pkg/apis/BUILD` file, these lines would be replaced with
```python
        "//staging/src/k8s.io/apimachinery/pkg/apis:all-srcs",
```

and then any subsequent changes would be isolated from this top-level BUILD file.

Reply:
Note that I did something similar for the top-level `BUILD` file in kubernetes/kubernetes; each top-level directory has a `BUILD` file, so the `all-srcs` rule at top-level is simple and isolated from changes:
https://github.com/kubernetes/kubernetes/blob/d0c3b708026f681f7467d36dcf9696f89ccfcfab/build/root/BUILD.root#L78-L97

Reply:
Thank you Jeff!